src/main/java/de/tum/in/www1/artemis/service/util/structureoraclegenerator/TypesDiff.java

```diff
@@ -0,0 +1,225 @@
+package de.tum.in.www1.artemis.service.util.structureoraclegenerator;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import spoon.reflect.declaration.*;
+import spoon.reflect.reference.CtTypeReference;
+
+/**
+ * This class represents the so-called diff of the solution type and the template type. 
+ * The solution type is a fully defined type found in the solution of a programming exercise.
+ * The template type its is counterpart found in the template of the same exercise.
+ * It is clear that the template type misses some of the structural elements of the solution type.
+ * This diff contains information on these structural elements.
+ * The ones the diff currently handles is:
+ * - Class name,
+ * - Package name,
+ * - Interface stereotype,
+ * - Enum stereotype,
+ * - Abstract modifier,
+ * - Superclass name,
+ * - Superinterfaces names,
+ * - Methods.
+ */
+public class TypesDiff {
+    private CtType<?> solutionType;
+    private CtType<?> templateType;
+    protected boolean isTemplateNull;
+    protected String name;
+	protected String packageName;
+	protected boolean isInterface;
+	protected boolean isEnum;
+	protected boolean isAbstract;
+	protected String superClassName;
+	protected Set<CtTypeReference<?>> superInterfaces;
+	protected Set<CtMethod<?>> methods;
+	protected boolean typesEqual;
+
+	public TypesDiff(CtType<?> solutionType, CtType<?> templateType) {
+	    this.solutionType = solutionType;
+	    this.templateType = templateType;
+        this.isTemplateNull = (templateType == null);
+        this.name = generateName();
+		this.packageName = generatePackageName();
+		this.isInterface = generateInterfaceStereotype();
+		this.isEnum = generateEnumStereotype();
+		this.isAbstract = generateAbstractModifier();
+		this.superClassName = generateSuperClassName();
+		this.superInterfaces = generateSuperInterfaces();
+		this.methods = generateMethodsDiff();
+		this.typesEqual = areTypesEqual();
+    }
+
+    /**
+     * This method gets the type name of both the solution and the template type. We do not talk about the diff here,
+     * since both the solution and the template type are supposed to have the same name.
+     * @return The name of the solution and template type.
+     */
+	private String generateName() {
```

Comment:
Hm this is strange, because IntelliJ displays the right identation and when viewing the code on GitHub it also gets displayed right. Only the pull request messes this up somehow. 🤔